WalletWasabi.Fluent/ViewModels/Settings/GeneralSettingsTabViewModel.cs

```diff
@@ -43,14 +36,18 @@ public GeneralSettingsTabViewModel()
 			? (FeeDisplayFormat)Services.UiConfig.FeeDisplayFormat
 			: FeeDisplayFormat.Satoshis;
 
+		ChangeTheme =
+			ReactiveCommand.CreateFromTask(async () =>
+			{
+				var light = DarkModeEnabled ? Theme.Dark : Theme.Light;
+				Services.UiConfig.DarkModeEnabled = DarkModeEnabled;
+				await MainViewModel.Instance.CompactDialogScreen.NavigateDialogAsync(new ThemeChangeViewModel(light));
+			});
+
 		this.WhenAnyValue(x => x.DarkModeEnabled)
 			.Skip(1)
-			.Subscribe(
-				x =>
-				{
-					Services.UiConfig.DarkModeEnabled = x;
-					Navigate(NavigationTarget.CompactDialogScreen).To(new ThemeChangeViewModel(x ? Theme.Dark : Theme.Light));
-				});
+			.Select(b => Unit.Default)
+			.InvokeCommand(ChangeTheme);
```

Comment:
Why was this required?

Reply:
This is because the previous `Navigate` call (method that is part of `RoutableViewModel`) was **synchronous**, whereas the in the new code I had to use `MainViewModel.Instance.CompactDialogScreen.NavigateDialogAsync`, that returns a `Task`. We should never subscribe Tasks.

Take this as an example:

```
observable.Subscribe(async x => await Something());
```

If you do that, you are getting the classic **async void** problem. Nobody will await that `Task`, and that's usually a big problem. Exceptions will become wild in such situations!

Example:

⛔**AVOID THIS**

```
this
	.WhenAnyValue(x => x.IsChecked)
	.Subscribe(async isChecked => await navigation.Navigate("Summary"));
```

✅ THIS IS OK

```
NavigateToSummary = ReactiveCommand.CreateFromTask( () => navigation.Navigate("Summary"));
this
	.WhenAnyValue(x => x.IsChecked)
	.Select(x => Unit.Default)
	.InvokeCommand(NavigateToSummary);
```

✅ THIS IS ALSO OK

```
this.WhenAnyValue(x => x.IsChecked)
  .SelectMany(_ => navigation.Navigate("summary"))
  .Subscribe();
```